rfcs/test262_integration.md

```diff
@@ -0,0 +1,70 @@
+# RFC 229: Integrating Test262
+
+## Summary
+
+This RFC proposes integrating the Test262 (ECMAScript) test suite into WPT. Due to its large size (>50,000 files), tests will be fetched on-demand rather than vendored into the repository. Execution will be opt-in via a new `--test262` flag. This provides a unified way to run this conformance suite while protecting the WPT repository from bloat and performance degradation.
```

Comment:
I strongly prefer just vendoring the test262 files directly.

Although that number is large it's about a 50% increase on the current number of files in the repo, so it's not unmanagable. 

Doing the vendoring directly means that there's no novel requirements for wpt consumers who want to run these tests (e.g. in vendor repos), apart from selecting the new test type.

I'd also propose a path like `third_party/test262` so that it's at least clear that this is not code that lives in wpt. I do think we're likely to run into cases where people are surprised when they make changes to this directory and they aren't upstreamed; at least the `third_party` convention provides a clue that this might not work.

Reply:
It looks like a shallow clone of Test262 is only 16M in the `.git/objects/` directory, so I think that's about how much WPT would grow, which seems OK.